SparkPost/2web2ui
# FE-726 - Add Content to Icon Buttons and Make Said Buttons Keyboard Accessible

[FE-726](https://jira.int.messagesystems.com/browse/FE-726)

### What Changed
- Added screen reader only content to many icon buttons in the navigation
- Updated markup to use `<button>` elements instead of anchor tags
- Added new `button-reset` Sass mixin to reset button styles
- Updating spacing between elements for consistency
- Updated failing tests
- Added `data-id` attributes to interactive items in the top navigation
- Added new `<IconButton/>` component with documentation and supporting unit tests
- Upgraded Storybook to version `5.2.6`
- Added documentation for many 2web2ui components:
  * `ActionPopover`
  * `Callout`
  * `LabelledValue`
  * `Loading`
  * `Note`
  * `SparkPostLogo`
  * `ButtonWrapper`
  * `IconButton`
  * `CopyField`
  * `DatePicker`
  * `DateSelector`
  * `MultiEmailField`
  * `DownloadLink`
  * `ExternalLink`
  * `PageLink`
  * `Empty`
  * `Loading`
  * `FocusContainer`
  * ...and an introduction page
- Added new build NPM script to eventually allow deployment

### How To Test
- Using a screen reader (or not!), verify that all buttons in the navigation have content
   - With a screen reader, you can use Voiceover, focus on the elements to verify -> https://www.apple.com/voiceover/info/guide/_1124.html
   - Without a screen reader, inspect the DOM to verify that all buttons have content even if it isn't visible
- Verify that all buttons within the navigation are interactive using the keyboard only - **hint:** use tab and shift+tab
- Run `npm install && npm run storybook-dev` to view the updated components and updated StoryBook documentation site

### To Do
